### PR TITLE
Chore: Create an `enum` for rule category.

### PR DESCRIPTION
**Pull request by sarvaje**
Chore: Remove `fixable` field.

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonar)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/developer-guide/contributing/pull-requests.html#commitmessages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

This PR create an `enum` to the category, so all the rules will have the same category with the same capitalization.

This PR also remove the rule property fixable because we are not using it and we are not going to use it shortly.

### CODE REVIEW

File: src/lib/types/rules.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
+import { Category } from '../enums/category';
+
+export type MetadataDocs = {
+    category: Category;
```
Comment (molant) [inline src/lib/types/rules.ts]: Maybe this field should be optional? I'm thinking about the rules that are outside sonar that might have a different category.
Another option will be to have `Category | string` and manually enforce a type when a rule is in core?

Comment (sarvaje) [reply src/lib/types/rules.ts]: How do you enforce a type when a rule is in the core?

**Comment by sarvaje** [comment]
@molant `category` is now optional and conflicts resolved

**Comment by alrra** [comment]
> FIxing typo

I've fixed commit message. :P

**Comment by molant** [comment]
> I've fixed the commit message. :P

I was planning in squashing...

**Pull request closed without merging**